SAT/Z3Interfacing.cpp

```diff
@@ -115,17 +115,12 @@ Z3Interfacing::Z3Interfacing(SAT2FO& s2f, bool showZ3, bool unsatCoresForAssumpt
   _config(),
   _context(_config),
   _solver(_context),
-  _model((STATEMENTS_TO_EXPRESSION(
-            BYPASSING_ALLOCATOR;
-            _solver.check();
-          ),
-         _solver.get_model())),
+  _model((STATEMENTS_TO_EXPRESSION(_solver.check();), _solver.get_model())),
```

Comment:
I think we now could go to
```
_model(_solver.check(),_solver.get_model()),
```
and delete the magical macro `STATEMENTS_TO_EXPRESSION`. What do you think?

Reply:
Yes - I saw this and scratched my head a few times, decided I would come back to it when I was more certain. If you think it should go too, I'm happy.

Reply:
Please!